P202-Training.Domain/UsersService.cs

```diff
@@ -35,24 +37,25 @@ public IList<User> ListUsers()
             return mapUser;
         }
 
-        public User ReadUser(int userId)
+        public User ReadUser(User user)
```

Comment:
Same here, the user ID should be enough for getting the user and in that case you could avoid unnecessary mappings